src/modules/esl-traversing-query/core/esl-traversing-query.ts

```diff
@@ -94,21 +94,22 @@ export class TraversingQuery {
     return uniq(result);
   }
 
-  static traverse(query: string, findFirst: boolean, base?: Element) {
+  static traverse(query: string, findFirst: boolean, base?: Element | null, scope: Element | null | Document = document) {
+    if (scope === null) throw new Error();
```

Comment:
In case we throw an error - there is no reason to allow null value at all

Reply:
I'd recommend restricting type and getting rid of an error (the user will have it anyway if he passed a null)
From an API point of view, we are also do not allow undefined or null scope, it is the required field for TraversingQuery
`base` element accepts null just for the purpose of being searched using query without base, e.g. queries like `#id::next` and etc.